thavalon-webapp/src/utils/GameSocket.tsx

```diff
@@ -0,0 +1,61 @@
+interface PingResponse {
+    result: boolean,
+    errorMessage: string, // only populated if result is false
+}
+
+/**
+ * The class containing the underlying websocket for playing the game.
+ * Will contain a websocket, that can either be retrieved or optionally made.
+ */
+export class GameSocket {
+    // the underlying gamesocket instance
+    private static instance: GameSocket;
+    // the instantiated instance of the underlying websocket
+    private websocket: WebSocket;
+
+    // create the underlying websocket in the constructor
+    private constructor(socketUrl: string) {
+        this.websocket = new WebSocket(socketUrl);
+        this.websocket.onopen = this.socketOnOpen;
+        this.websocket.onmessage = this.socketOnMessage;
+        this.websocket.onclose = this.socketOnClose
+        this.websocket.onerror = this.socketOnError;

```

Comment:
Do these callbacks need to be bound to `this` or does typescript do that automatically?

Reply:
From testing, looks like typescript automatically does that.